framework/src/auxkernels/FunctorCoordinatesFunctionAux.C

```diff
@@ -0,0 +1,80 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FunctorCoordinatesFunctionAux.h"
+#include "Function.h"
+
+registerMooseObject("MooseApp", FunctorCoordinatesFunctionAux);
+
+InputParameters
+FunctorCoordinatesFunctionAux::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.addClassDescription(
+      "Auxiliary Kernel that creates and updates a field variable by "
+      "sampling a function with functors (variables, functions, others) as the coordinates.");
+  params.addRequiredParam<FunctionName>("function", "The function to use as the value");
+  params.addRequiredParam<MooseFunctorName>(
+      "x_functor", "The functor to use for the X coordinate function argument");
+  params.addRequiredParam<MooseFunctorName>(
+      "y_functor", "The functor to use for the Y coordinate function argument");
+  params.addRequiredParam<MooseFunctorName>(
+      "z_functor", "The functor to use for the Z coordinate function argument");
+  params.addRequiredParam<MooseFunctorName>("t_functor",
+                                            "The functor to use for the time function argument");
+  params.addParam<MooseFunctorName>("factor", 1, "A factor to apply on the functor");
+
+  return params;
+}
+
+FunctorCoordinatesFunctionAux::FunctorCoordinatesFunctionAux(const InputParameters & parameters)
+  : AuxKernel(parameters),
+    _func(getFunction("function")),
+    _x_functor(getFunctor<Real>("x_functor")),
+    _y_functor(getFunctor<Real>("y_functor")),
+    _z_functor(getFunctor<Real>("z_functor")),
+    _t_functor(getFunctor<Real>("t_functor")),
+    _factor(getFunctor<Real>("factor")),
+    _is_fe(dynamic_cast<MooseVariableFE<Real> *>(&_var))
+{
+  if (!_is_fe && !dynamic_cast<MooseVariableFV<Real> *>(&_var))
```

Comment:
This should be updated with the new linear finite volume variables. They support functor evaluation

Reply:
addressed

Reply:
it s not really about evaluation here though, it s more about "what can be set by an auxkernel" I dont know that we need this check even since we should complain upstream about an auxkernel being used on a non-real valued variable